### PR TITLE
Remove empty relay state from redirect url

### PR DESCRIPTION
**Pull request by clemstoquart**
Fix https://github.com/spring-projects/spring-security/issues/7640

### CODE REVIEW

**Comment by rwinch** [comment]
Thanks for the PR @clemstoquart! Could you please add a test for this?

**Comment by clemstoquart** [comment]
@rwinch Yes, I'll do that tomorrow ;)

**Comment by clemstoquart** [comment]
Hi @rwinch 

I've added some tests.

File: saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationRequestFilterTests.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.security.saml2.Saml2Exception;
+import org.springframework.security.saml2.credentials.Saml2X509Credential;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistration;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistrationRepository;
+
+import static java.nio.charset.StandardCharsets.*;
+import static org.mockito.Mockito.*;
+import static org.springframework.security.saml2.credentials.Saml2X509Credential.Saml2X509CredentialType.*;
```
Comment (fhanik) [inline saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationRequestFilterTests.java]: Can we expand these imports please.

File: saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationRequestFilterTests.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+				.thenReturn(relyingPartyRegistration);
+
+		request.setParameter("RelayState", "my-relay-state");
+
+		filter.doFilterInternal(request, response, filterChain);
+
+		Assert.assertTrue(response.getHeader("Location").contains("RelayState=my-relay-state"));
+	}
+
+	private static X509Certificate certificate(String cert) {
```
Comment (fhanik) [inline saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationRequestFilterTests.java]: We typically break out test support data into a separate class prefixed with `TestNameOfDataWeProvide`

For example you can take a look at: [TestSaml2Credentials](https://github.com/fhanik/spring-security/blob/feature/saml2_login_configurer_tests/config/src/test/java/org/springframework/security/config/annotation/web/configurers/saml2/TestSaml2Credentials.java)

Comment (clemstoquart) [reply saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationRequestFilterTests.java]: Do you think it's ok to reuse data from TestSaml2X509Credentials by making them public (or moving the class) ?

File: saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationRequestFilterTests.java (hunk 1 above)
Comment (clemstoquart) [reply saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationRequestFilterTests.java]: Sure. It's done.

File: saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationRequestFilterTests.java (hunk 2 above)
Comment (fhanik) [reply saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationRequestFilterTests.java]: For now, if you include a package private class, we can consolidate test support later.

Comment (clemstoquart) [reply saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationRequestFilterTests.java]: Thanks for the review.

I've moved test data to another class.

**Pull request merged**